api_tests/registrations/views/test_registration_list.py

```diff
@@ -1913,6 +1913,150 @@ def test_bulk_update_embargo_logged_in_contrib(
         assert registration_two.is_pending_embargo_termination is True
 
 
+class TestRegistrationContributors(ApiTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.user = AuthUserFactory()
+
+        self.project = ProjectFactory(is_public=False, creator=self.user)
+        self.registration_project = RegistrationFactory(
+            creator=self.user, project=self.project)
+
+        self.public_project = ProjectFactory(is_public=True, creator=self.user)
+        self.public_registration_project = RegistrationFactory(
+            creator=self.user, project=self.public_project, is_public=True)
+        self.contributor = AuthUserFactory()
+    
+    def add_contributor_request(self, auth_user, contributor, permission='write', expect_errors=False):
+        url = f'/{API_BASE}registrations/{self.public_registration_project._id}/contributors/'
+        attrs = {
+            "permission": permission,
+            "bibliographic": True
+        }
+        relationships = {
+            "users": {
+                "data": {
+                    "type": "users",
+                    "id": contributor._id
+                }
+            }
+        }
+        payload = {'data': {'attributes': attrs, 'relationships': relationships, 'type': 'contributors'}}
```

Comment:
just a small thing but I would like everything to be indented like the `relationships` dictionary.